integration-tests/modules/__tests__/product/workflows/product-options-removal-validation.spec.ts

```diff
@@ -0,0 +1,365 @@
+          const product = await productModule.createProducts({
+            title: "Test Product No Variants",
+            options: [
+              {
+                title: "Color",
+                values: ["Red", "Blue"],
+              },
+            ],
+          })
+
+          const optionId = product.options[0].id
+
+          // Remove the option - should succeed
+          await productModule.removeProductOptionFromProduct({
+            product_id: product.id,
+            product_option_id: optionId,
+          })
+
+          const updatedProduct = await productModule.retrieveProduct(
+            product.id,
+            { relations: ["options"] }
+          )
+          expect(updatedProduct.options).toHaveLength(0)
+        })
+      })
+
+      describe("addProductOptionToProduct with value_ids filter", () => {
+        it("should throw error when trying to remove ProductOptionValue used by variant", async () => {
+          // Create product with option having multiple values
+          const product = await productModule.createProducts({
+            title: "Test Product",
+            options: [
+              {
+                title: "Size",
+                values: ["S", "M", "L", "XL"],
+              },
+            ],
+          })
+
+          const option = product.options[0]
+          const valueS = option.values.find((v) => v.value === "S")!
+          const valueM = option.values.find((v) => v.value === "M")!
+
+          // Create variant using value "S"
+          await productModule.createProductVariants({
+            product_id: product.id,
+            title: "Variant S",
+            options: { Size: "S" },
+          })
+
+          // Attempt to update option to exclude value "S" (keeping only M, L, XL)
+          const remainingValueIds = option.values
+            .filter((v) => v.value !== "S")
+            .map((v) => v.id)
+
+          await expect(
+            productModule.addProductOptionToProduct({
+              product_id: product.id,
+              product_option_id: option.id,
+              product_option_value_ids: remainingValueIds,
+            })
+          ).rejects.toThrow(MedusaError)
+
+          await expect(
+            productModule.addProductOptionToProduct({
+              product_id: product.id,
+              product_option_id: option.id,
+              product_option_value_ids: remainingValueIds,
+            })
+          ).rejects.toThrow(
+            "Cannot unassign option values from product because the following variant(s) are using it"
+          )
+
+          try {
+            await productModule.addProductOptionToProduct({
+              product_id: product.id,
+              product_option_id: option.id,
+              product_option_value_ids: remainingValueIds,
+            })
+          } catch (error) {
+            expect((error as MedusaError).message).toContain("Variant S")
+          }
```

Comment:
todo: try catch are generally not a good practice, if it doesnt fail, the expectation never run and the test would succeed. two options, either adding an expectation in the try such as
```ts
try {
  await productModule.addProductOptionToProduct({
    product_id: product.id,
    product_option_id: option.id,
    product_option_value_ids: remainingValueIds,
  })
  expect(true).toBe(false)
} catch (e) {
  expect((error as MedusaError).message).toContain("Variant S")
}
```

or alternatively

```ts
const res = await productModule.addProductOptionToProduct({
  product_id: product.id,
  product_option_id: option.id,
  product_option_value_ids: remainingValueIds,
}).catch(e => e)
expect((error as MedusaError).message).toContain("Variant S")
```